File: VB/VB/WinWebSolution.Module/MyDetailsWorker.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Actions

Namespace WinWebSolution.Module
	Public Class MyDetailsWorker
		Inherits ViewController
		Private saStaticTestAction As SimpleAction
		Private components As System.ComponentModel.IContainer
		Private dynamicTestActionCore As SimpleAction
		Public Sub New()
			'Static action
			InitializeComponent()
			RegisterActions(components)
			'Dynamic action
			dynamicTestActionCore = New SimpleAction(Me, "DynamicTestAction", PredefinedCategory.RecordEdit)
			dynamicTestActionCore.Caption = "Dynamic Test Action"
		End Sub
		Protected Overrides Overloads Sub OnFrameAssigned()
			MyBase.OnFrameAssigned()
			If Application IsNot Nothing Then
				Dim os As ObjectSpace = Application.CreateObjectSpace()
				dynamicTestActionCore.TargetViewId = Application.GetDetailViewId(SecuritySystem.UserType)
				dynamicTestActionCore.TargetObjectsCriteria = CriteriaOperator.Parse("?=?", os.GetKeyPropertyName(SecuritySystem.UserType), SecuritySystem.CurrentUserId).ToString()
			End If
		End Sub
		Private Sub InitializeComponent()
			Me.components = New System.ComponentModel.Container()
			Me.saStaticTestAction = New DevExpress.ExpressApp.Actions.SimpleAction(Me.components)
			' 
			' saStaticTestAction
			' 
			Me.saStaticTestAction.Caption = "Static Test Action"
			Me.saStaticTestAction.Category = "RecordEdit"
			Me.saStaticTestAction.Id = "StaticTestAction"
			Me.saStaticTestAction.Tag = Nothing
			Me.saStaticTestAction.TargetObjectsCriteria = "Oid='@CurrentUserID'"
			Me.saStaticTestAction.TargetViewId = "SimpleUser_DetailView"
			Me.saStaticTestAction.TypeOfView = Nothing

		End Sub
	End Class
End Namespace

File: CS/E2033/WinWebSolution.Module/MyDetailsWorker.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Actions;

namespace WinWebSolution.Module {
    public class MyDetailsWorker : ViewController {
        private SimpleAction saStaticTestAction;
        private System.ComponentModel.IContainer components;
        private SimpleAction dynamicTestActionCore;
        public MyDetailsWorker() {
            //Static action
            InitializeComponent();
            RegisterActions(components);
            //Dynamic action
            dynamicTestActionCore = new SimpleAction(this, "DynamicTestAction", PredefinedCategory.RecordEdit);
            dynamicTestActionCore.Caption = "Dynamic Test Action";
        }
        protected override void OnFrameAssigned() {
            base.OnFrameAssigned();
            if (Application != null) {
                ObjectSpace os = Application.CreateObjectSpace();
                dynamicTestActionCore.TargetViewId = Application.GetDetailViewId(SecuritySystem.UserType);
                dynamicTestActionCore.TargetObjectsCriteria = CriteriaOperator.Parse("?=?", os.GetKeyPropertyName(SecuritySystem.UserType), SecuritySystem.CurrentUserId).ToString();
            }
        }
        private void InitializeComponent() {
            this.components = new System.ComponentModel.Container();
            this.saStaticTestAction = new DevExpress.ExpressApp.Actions.SimpleAction(this.components);
            // 
            // saStaticTestAction
            // 
            this.saStaticTestAction.Caption = "Static Test Action";
            this.saStaticTestAction.Category = "RecordEdit";
            this.saStaticTestAction.Id = "StaticTestAction";
            this.saStaticTestAction.Tag = null;
            this.saStaticTestAction.TargetObjectsCriteria = "Oid=\'@CurrentUserID\'";
            this.saStaticTestAction.TargetViewId = "SimpleUser_DetailView";
            this.saStaticTestAction.TypeOfView = null;

        }
    }
}